### PR TITLE
[feat ] responsive header

### PR DESCRIPTION
**Pull request by DLavin23**
@hatchloyalty/engineering 

image

<hr/>

image

<hr/>

image

<hr/>

### CODE REVIEW

File: src/components/__tests__/header.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { shallow, ShallowWrapper } from 'enzyme'
-import { Link } from 'gatsby'
 import 'jest' // tslint:disable-line no-import-side-effect
 import * as React from 'react'
 import Headroom from 'react-headroom'
 import Header, { IHeaderProps } from '../header'
+import Nav from '../nav'
```
Comment (hstrowd) [inline src/components/__tests__/header.tsx]: /nonblocker We should probably discuss and strive to standardize when we use relative and absolute imports.

File: src/components/__tests__/header.tsx [hunk 2]
```diff
@@ -46,11 +50,11 @@ describe('Header', () => {
     })
 
     it('renders each specified navigation link', async () => {
-      expect(mountedComponent.find(Link).length).toEqual(
-        1 + props.navLinks.length
+      expect(mountedComponent.find(Nav.Link).length).toEqual(
+        0 + props.navLinks.length
```
Comment (hstrowd) [inline src/components/__tests__/header.tsx]: We can probably drop the `0 +` entirely.

File: src/components/header.tsx [hunk 3]
```diff
@@ -1,77 +1,114 @@
...
+              isVisible={this.state.showNav}
+              onClick={this.toggleNav}
+            >
+              {this.state.showNav ? <X size={28} color="#fff"/> : <Menu size={28} color="#484848"/>}
+            </Nav.Toggle>
 
-            <Nav>{links}</Nav>
-          </Wrapper>
+            <Nav isVisible={this.state.showNav}>
+              <Nav.List isVisible={this.state.showNav}>
```
Comment (hstrowd) [inline src/components/header.tsx]: It catches me a bit off guard that this `isVisible` attribute needs to be set on both of thees elements. Is it not sufficient to control the visibility entirely at the `Nav` component level?

File: src/components/header.tsx [hunk 4]
```diff
@@ -1,77 +1,114 @@
...
+    showNav: false,
+  }
+
+  toggleNav = () => {
+    this.setState({ showNav: !this.state.showNav })
+  }
+
   render() {
     const { siteTitle, navLinks, pageBackground, pageColor } = this.props
+    const StickyHeader = this.state.showNav ? Box : Headroom
```
Comment (hstrowd) [inline src/components/header.tsx]: Could we use the `disable` prop on the Headroom component to accomplish this (i.e. set `disable` to true whenever `this.state.showNav` is true)?

File: src/components/header.tsx (hunk 3 above)
Comment (DLavin23) [reply src/components/header.tsx]: unfortunately, it is necessary.

File: src/components/header.tsx (hunk 4 above)
Comment (DLavin23) [reply src/components/header.tsx]: Nope. Tried that.

**Pull request merged**